File: base/src/main/java/de/whitefrog/frogr/exception/TypeMismatchException.java
package de.whitefrog.frogr.exception;

import org.neo4j.graphdb.Label;
import org.neo4j.graphdb.Node;

/**
 * Thrown when a model is passed with the wrong type information.
 */
public class TypeMismatchException extends FrogrException {
  Node node;
  Label label;

  public TypeMismatchException(Node node, Label label) {
    super("The expected label \"" + label + "\" was not found on " + node);
    this.node = node;
    this.label = label;
  }
}

File: base/src/main/java/de/whitefrog/frogr/exception/TypeMismatchException.kt
package de.whitefrog.frogr.exception

import org.neo4j.graphdb.Label
import org.neo4j.graphdb.Node

/**
 * Thrown when a model is passed with the wrong type information.
 */
class TypeMismatchException(internal var node: Node, internal var label: Label) : 
  FrogrException("The expected label \"$label\" was not found on $node")
